proto/src/rr/resource.rs

```diff
@@ -294,11 +294,16 @@ impl<'r> BinDecodable<'r> for Record {
         //                with a zero TTL to prohibit caching.  Zero values can
         //                also be used for extremely volatile data.
         // note: u32 seems more accurate given that it can only be positive
-        let ttl: u32 = decoder.read_u32()?;
+        let ttl: u32 = decoder.read_u32()?.unverified();
 
         // RDLENGTH        an unsigned 16 bit integer that specifies the length in
         //                octets of the RDATA field.
-        let rd_length: u16 = decoder.read_u16()?;
+        let rd_length: u16 = decoder
+            .read_u16()?
+            .verify_unwrap(|u| (*u as usize) <= decoder.len())
```

Comment:
we're down catch a potential case where the rdata length is longer than the DNS packet. I don't think it was possible to exploit it easily (except through the other rdata_length usages) b/c if it was too long we'd eventually fail to pop enough bytes but, still this is better.